### PR TITLE
F/update codesniffer 3.x

### PR DESCRIPTION
**Pull request by jtyost2**
Updates to include the 3.x version of the Code Sniffer.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
     "cakephp/plugin-installer": "*"
   },
   "require-dev": {
-    "phpunit/phpunit": "4.1.*"
+    "phpunit/phpunit": "4.1.*",
+    "loadsys/loadsys_codesniffer": "~3.0"
```
Comment (beporter) [inline composer.json]: This isn't required. The **composer.json** is only used during `create-project`. The **composer.json.template** will be transformed into the composer.json "proper" for the generated app, so the `.template` version is the only thing we need to keep up to date. I'm reminded of a comment I made... (https://github.com/loadsys/CakePHP-Skeleton/issues/70#issuecomment-72271709)

Comment (beporter) [inline composer.json]: That is, unless the intent is to run the codesniffer **on** the skeleton project itself, perhaps via Travis...

Comment (jtyost2) [inline composer.json]: I added it due to these lines: https://github.com/loadsys/CakePHP-Skeleton/pull/117/files#diff-04c6e90faac2675aa89e2176d2eec7d8L150

Comment (beporter) [inline composer.json]: Sure, a reasonable action.

I think those notes were put in place before we had a `composer.json.template` version of the file though.

**Comment by jtyost2** [comment]
Removed the codesniffer from the `composer.json` as discussed.

**Pull request merged**